headers/TextureManager.hpp

```diff
@@ -1,6 +1,8 @@
 #pragma once
 
+#include <iostream>
```

Comment:
`TextureManager.hpp` doesn't use anything contained in `iostream`, it would be better to move this include into the implementation file `TextureManager.hpp`. This way the implementation is not relying on the header to provide `iostream`.